internals/daemon/api_checks.go

```diff
@@ -66,3 +67,34 @@ func v1GetChecks(c *Command, r *http.Request, _ *UserState) Response {
 	}
 	return SyncResponse(infos)
 }
+
+func v1PostChecks(c *Command, r *http.Request, _ *UserState) Response {
+	var payload struct {
+		Action string `json:"action"`
+		Check  string `json:"check"`
+	}
+	decoder := json.NewDecoder(r.Body)
+	if err := decoder.Decode(&payload); err != nil {
+		return BadRequest("cannot decode data from request body: %v", err)
+	}
+	switch payload.Action {
+	case "run":
+		if payload.Check == "" {
+			return BadRequest("no check to %s provided", payload.Action)
+		}
+
+		check, ok := c.d.overlord.PlanManager().Plan().Checks[payload.Check]
+		if !ok {
+			return SyncResponse("check not found")
```

Comment:
Should be an error response, see other similar code. `return NotFound("cannot find check with name %q", payload.Check)`